convert2rhel/main.py

```diff
@@ -136,6 +136,7 @@ def main_locked():
     initialize_file_logging("convert2rhel.log", logger_module.LOG_DIR)
 
     try:
+        logger_module.ConversionStage.set_stage("prepare")
```

Comment:
What about using `ConversionPhase` for this? I think it could be really good to map the phases in that class to the `ConversionStage` you created

Reply:
@r0x0d That's a good point. Though that made me dig down a rabbit hole of refactoring to be a bit nicer to use for more scenarios. What do you think of this?

```py
class ConversionPhase:
    """Conversion phase to hold name and logging-friendly name.
    """

    def __init__(self, name, log_name=None):  # type: (str, str|None) -> None
        self.name = name
        self.log_name = log_name

    def __str__(self):
        return self.log_name

    def __repr__(self):
        return self.name

    def __hash__(self) -> int:
        return hash(self.name)


class ConversionPhases(dict):
    """During conversion we will be in different states depending on
    where we are in the execution. This class establishes the different phases
    that we have as well as what the current phase is set to.
    """
    _phases = [
        ConversionPhase(name="POST_CLI", log_name="Prepare"),
        ConversionPhase(name="PRE_PONR_CHANGES", log_name="Prepare")
    ]

    current_phase = None  # type: ConversionPhase|None

    def __getitem__(self, key):  # type: (str) -> ConversionPhase|None
        return next(phase for phase in self._phases if phase.name == key )

    @classmethod
    def set_stage(cls, phase):  # type: (str) -> None
        if phase is None or phase in cls._phases:
            cls.current_phase = cls._phases[phase]
        raise NotImplementedError("The stage {} is not implemented in the ConversionStage class".format(phase))

test = ConversionPhases()

print(test["POST_CLI"].__repr__)
```
```
vscode(container):/workspaces/convert2rhel$ /bin/python3 /workspaces/convert2rhel/convert2rhel/utils/test.py
<bound method ConversionPhase.__repr__ of POST_CLI>
```

Reply:
Hmmm, this work as well, but it is easy to introduce an error if we miss the correct naming in other parts of the code or introduce a new phase and forget about it. 

I was thinking in something more simple, like this: 

```python
class ConversionPhase:
    POST_CLI = 1
    # PONR means Point Of No Return
    PRE_PONR_CHANGES = 2
    # Phase to exit the Analyze SubCommand early
    ANALYZE_EXIT = 3
    POST_PONR_CHANGES = 4

class ConversionStage:
    stages = {
        ConversionPhase.PRE_PONR_CHANGES: "Prepare",
        ConversionPhase.POST_PONR_CHANGES: "Final",
        ...
    }
    current = None  # type: str|None

    @classmethod
    def set_stage(cls, stage):  # type: (str) -> None
        if stage == None or stage in cls.stages:
            cls.current = cls.stages[stage]
        raise NotImplementedError("The stage {} is not implemented in the ConversionStage class".format(stage))
```

Then, when we initialize the `ConversionStage`, we could just simply: 

```python
logger_module.ConversionStage.set_stage(ConversionPhase.PRE_PONR_CHANGES)
```

And then we have it mapped everywhere that `PRE_PONR_CHANGES` matches the `Prepare` or something else. The interface will still look clean enough in the new class, and we will make sure that if someone changes the meaning for any of those phases, we will catch somewhere as well. 

Since our use case is so small for now, I think that reusing the phases from `ConversionPhase` should be good enough.

Reply:
I think our `ConversionPhase` class should hold the information where we are in the process at all times, being our source of truth for many different operations. It is true that logging phases may differ from execution phases, but if that's really the case, then we could come up with an intermediate class that will hold the logging phases for us, tied to the conversion phase if necessary.

Reply:
Ah I didn't see this as I got hyperfocused on overcomplicating the solution... but anyhow I did change it now to be a bit safer but in a way that means we still rely on strings. But given exceptions being raised we should catch it in tests if we do a mistake